x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/selectors.ts

```diff
@@ -72,8 +107,38 @@ export const urlFromNewPageIndexParam: (
   state: AlertListState
 ) => (newPageIndex: number) => string = state => {
   return newPageIndex => {
-    const urlPaginationData = paginationDataFromUrl(state);
+    const urlPaginationData: AlertingIndexUIQueryParams = { ...uiQueryParams(state) };
     urlPaginationData.page_index = newPageIndex.toString();
     return '?' + qs.stringify(urlPaginationData);
   };
 };
+
+/**
+ * Returns a url like the current one, but with a new alert id.
+ */
+export const urlWithSelectedAlert: (
```

Comment:
Should these be named something like `urlSearchParams*` to be more reflective of what it is returning?

Reply:
I view this as returning a relative URL. It is a string, beginning with `?` and so it fits the technical def: https://tools.ietf.org/html/rfc1808#section-5

The intent is that these will be used with `history.pushState` or the `href` attribute of an `a` tag. These accept relative URLs and treat them as absolute after combining them with the base URL of the page.

Maybe I could explain the above in a comment. With all that being said, do you still think I should go for a diff name?